common/http_test.cpp

```diff
@@ -1824,6 +1824,58 @@ TEST(HttpsTest, WrongSelfSignedCertificateError) {
 	EXPECT_FALSE(client_hit_body);
 }
 
+TEST(HttpsTest, CertificationWithWrongHostName) {
+	GTEST_SKIP() << "MEN-6788";
+
+	TestEventLoop loop;
+
+	bool client_hit_header {false};
+	bool client_hit_body {false};
+
+	mendertesting::TemporaryDirectory tmpdir;
+	string script = R"(#! /bin/sh
+	  exec openssl s_server -www )";
+	script += " -key server.localhost.key";
+	script += " -cert server.localhost.crt";
+	script += " -accept " TEST_PORT;
+
+	const string script_fname = tmpdir.Path() + "/test-script.sh";
+	{
+		std::ofstream os(script_fname.c_str(), std::ios::out);
+		os << script;
+	}
+	int ret = chmod(script_fname.c_str(), S_IRUSR | S_IWUSR | S_IXUSR);
+	ASSERT_EQ(ret, 0);
+	processes::Process server({script_fname});
+	auto err = server.Start();
+	ASSERT_EQ(err, error::NoError);
+	std::this_thread::sleep_for(std::chrono::seconds {1}); // Give the server a little time to setup
+
+	http::ClientConfig client_config {"server.localhost.crt"};
+	http::Client client(client_config, loop);
+	auto req = make_shared<http::OutgoingRequest>();
+	req->SetMethod(http::Method::GET);
+	// No hostname, therefore, should fail validation.
+	req->SetAddress("https://127.0.0.1:" TEST_PORT "/index.html");
+	err = client.AsyncCall(
+		req,
+		[&client_hit_header, &loop](http::ExpectedIncomingResponsePtr exp_resp) {
+			client_hit_header = true;
+			ASSERT_FALSE(exp_resp);
+			loop.Stop();
+		},
+		[&client_hit_body, &loop](http::ExpectedIncomingResponsePtr exp_resp) {
+			client_hit_body = true; // This should never happen
+			loop.Stop();
```

Comment:
Nit: I know this is copy and paste, but `loop.Stop()` should not be necessary here right?

Reply:
Correct, but I believe I've hit some of these while implementing things, and it terminates faster then. I don't consider it wrong, so I think I'll leave it.